Assignment01/step-by-step-python.md

```diff
@@ -0,0 +1,191 @@
+
+2. Click on `Send request` in the file to send a request to the API:
+
+   ![REST client](img/rest-client.png)
+
+3. The response of the request will be shown in a separate window on the right. It should be a response with HTTP
+   status code `200 OK` and the body should contain some random vehicle and owner-information:
+
+   ```json
+   HTTP/1.1 200 
+   Connection: keep-alive
+   Content-Type: application/json
+   Date: Wed, 16 Jun 2021 19:39:05 GMT
+   Keep-Alive: timeout=60
+   Transfer-Encoding: chunked
+   
+   {
+       "vehicle_id": "KZ-49-VX",
+       "brand": "Toyota",
+       "model": "Rav 4",
+       "owner_name": "Angelena Fairbairn",
+       "owner_email": "angelena.fairbairn@outlook.com"
+   }
+   ```
+
+4. Check the logging in the terminal window. It should look like this:
+
+   ```console
+   > uvicorn vehicle_registration:app --port 6002
+   INFO:     Started server process [29384]
+   INFO:     Waiting for application startup.
+   INFO:     Application startup complete.
+   INFO:     Uvicorn running on http://127.0.0.1:6002 (Press CTRL+C to quit)
+   INFO:     127.0.0.1:64855 - "GET /vehicleinfo/KZ-49-VX HTTP/1.1" 200 OK
+   ```
+
+## Step 2. Run the FineCollection service
+
+1. Make sure the VehicleRegistrationService service is running (result of step 1).
+
+1. Open a **new** terminal window in VS Code.
+
+   > You can do this by using the hotkey (``Ctrl-` `` on Windows, ``Shift-Ctrl-` `` on macOS) or clicking on
+   > the `+` button in the terminal window title bar:
+   > ![](img/terminal-new.png)
+
+1. Make sure the current folder is `python/FineCollectionService`.
+
+1. Run the command `pip install -r requirements.txt`
+
+1. Start the service using `uvicorn fine_collection:app --port 6001`.
+
+1. Open the file `python/FineCollectionService/test.http` in VS Code. The request in this file simulates sending a
+   detected speeding-violation to the FineCollectionService.
+
+2. Click on `Execute request` in the file to send a request to the API.
+
+3. The response of the request will be shown in a separate window on the right. It should be a response with HTTP
+   status code `200 OK` and no body.
+
+4. Check the logging in the terminal window. It should look like this:
+
+   ```console
+   ❯ uvicorn fine_collection:app --port 6001
+   INFO:     Started server process [3108]
+   INFO:     Waiting for application startup.
+   INFO:     Application startup complete.
+   INFO:     Uvicorn running on http://127.0.0.1:6001 (Press CTRL+C to quit)
+   INFO:     127.0.0.1:52778 - "POST /collectfine HTTP/1.1" 200 OK
+   ```
+
+## Step 3. Run the TrafficControl service
+
+1. Make sure the VehicleRegistrationService and FineCollectionService are running (results of step 1 and 2).
+
+1. Open a **new** terminal window in VS Code and make sure the current folder is `python/TrafficControlService`.
+
+1. Run the command `pip install -r requirements.txt`
+
+1. Start the service using `uvicorn traffic_control:app --port 6000`.
```

Comment:
Other than the VehicileRegistrationService, this command gives no output. When I run the HTTP request, it turns out the Python app is indeed running.

Reply:
I have a custom logging config in place. I can remove it. I'm getting a ton of duplicate message without it. 
Which would you prefer? No logging or duplicate messages?

Reply:
I'm going to take another look at the fastapi docs and see if I can properly fix this. It looks stupid 😅

Reply:
Fixed the logging with d0aa745d3ebd7b5dd9b04331217f9bd1443d3584